packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -330,16 +384,27 @@ export class ModularChangeFamily
 			const composedChange = fieldKind.changeHandler.rebaser.compose(
 				taggedChange1,
 				taggedChange2,
-				(child1, child2) =>
-					this.composeNodeChanges(
-						child1,
-						revision1,
-						child2,
-						revision2,
-						genId,
-						crossFieldTable,
-						revisionMetadata,
-					),
+				(child1, child2) => {
+					const taggedId1 =
+						child1 !== undefined
+							? tagChange(child1, taggedChange1.revision)
+							: undefined;
+
+					const taggedId2 =
+						child2 !== undefined
+							? tagChange(child2, taggedChange2.revision)
+							: undefined;
+
+					crossFieldTable.nodeIdPairs.push([taggedId1, taggedId2]);
+					if (child1 !== undefined && child2 !== undefined) {
```

Comment:
Is it important that we don't add child2 to crossFieldTable.nodeIds when child1 is undefined? The doc comment on crossFieldTable.nodeIds doesn't mention this kind of filtering.

Reply:
I think I get it: `crossFieldTable.nodeIds` is used to avoid re-computing a composition for a (child1, child2) pair. If we didn't have this `child1 !== undefined`, we would fail to compute a (child1, child2) composition if we had run into a (undefined, child2) composition first.

Reply:
Wait, but we should never run into both (undefined, child2) and (child1, child2) for the same child2. So I'm back to wondering why we need this `child1 !== undefined` check. Is it merely an optimization to avoid populating the map when the entries in it would only save us O(1) time?